.github/workflows/test-sample-go.yml

```diff
@@ -24,9 +24,9 @@ jobs:
         run: |
           KUSTOMIZATION_FILE_PATH="testdata/project-v4/config/default/kustomization.yaml"
           sed -i '25s/^#//' $KUSTOMIZATION_FILE_PATH

```

Comment:
@joelanford this one is blocked by #3937
I am waiting your review in #3937 and in the second commit here so that we can
- Get #3937 merged
- Rebase this one and get this one merged 
- Cherry-pick both and push a new 3.15.1 release